src/python/dxpy/scripts/dx.py

```diff
@@ -3229,10 +3229,26 @@ def run_body(args, executable, dest_proj, dest_path, preset_inputs=None, input_n
     # into the runtime systemRequirements
     requested_system_requirements = (requested_instance_type + requested_cluster_spec + requested_fpga_driver).as_dict()
 
-    # store runtime --instance-type-by-executable {executable:{entrypoint:{'instanceType':xxx}}} as systemRequirementsByExecutable 
+    if (args.instance_type and (args.instance_type_by_executable or cloned_system_requirements_by_executable)):
+        warning = BOLD("WARNING") + ": --instance-type argument {} may get overridden by\n".format(args.instance_type) 
```

Comment:
https://confluence.internal.dnanexus.com/pages/viewpage.action?spaceKey=TTN&title=dx+run+workflow-xxxx+--instance-type-by-executable+--delay-workspace-destruction 
4.3.1 says "If "dx run --clone fromId"  is called with "--instance-type Y" input argument and there's a chance this --instance-type argument will be overridden by a combination of "--instance-type-by-executable X" and fromId.mergedSystemRequirementsByExecutable, print "Warning: --instance-type argument <Y> may get overridden by --instance-type-by-executable argument <X> and <fromId> mergedSystemRequirementsByExecutable value of <>".

The `if` statement may insert a warning event where there's no --clone (i.e. where there's both instance_type and instance_type_by_executable) - this is not what's speced

Reply:
Thanks! @emiloslavsky now it is fixed:
```
➜ dx run applet-GZv3JP801fX89p7Pkkzbb9Pq --instance-type mem1_sdd2_v2_x4 --clone job-GYBfXBQ00PB77fyv39FF719Q --instance-type-by-executable \{\"applet-GZv3JP801fX89p7Pkkzbb9Pq\":\{\"mai
n\":\"mem1_sdd2_v2_x2\"\}\} -y 

WARNING: --instance-type argument: mem1_sdd2_v2_x4 may get overridden by
job-GYBfXBQ00PB77fyv39FF719Q mergedSystemRequirementsByExecutable:
{"applet-GZv3JP801fX89...7Pkkzbb9Pq": {"main": "mem1_sdd2_x2"}} and runtime
--instance-type-by-executable argument:{"applet-GZv3JP801fX89p7Pkkzbb9Pq":
{"main": "mem1_sdd2_v2_x2"}}

➜ dx run applet-GZv3JP801fX89p7Pkkzbb9Pq --instance-type mem1_sdd2_v2_x4 --clone job-GYBfXBQ00PB77fyv39FF719Q -y 
WARNING: --instance-type argument: mem1_sdd2_v2_x4 may get overridden by
job-GYBfXBQ00PB77fyv39FF719Q mergedSystemRequirementsByExecutable:
{"applet-GZv3JP801fX89...7Pkkzbb9Pq": {"main": "mem1_sdd2_x2"}}
```